cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -362,6 +362,17 @@ private void restartIfNecessary(NodeRef nodeRef, RestartContext restartContext)
             throw new UnforceableProblem("Error getting pod " + nodeRef.podName(), e);
         }
 
+        if (!isPodStuck(pod)) {
+            // We want to give pods chance to get ready before we try to connect to the or consider them for rolling.
+            // But only in case when they are not stuck as that would
+            LOGGER.debugCr(reconciliation, "Waiting for pod {} to become ready before checking its state", nodeRef.podName());
+            try {
+                await(isReady(pod), operationTimeoutMs, TimeUnit.MILLISECONDS, e -> new RuntimeException(e));
+            } catch (Exception e)   {
+                LOGGER.warnCr(reconciliation, "Pod {} is not ready. We will check if KafkaRoller can do anything about it.", nodeRef.podName());
```

Comment:
This message isn't entirely accurate: We can only conclude the pod is not ready is the await timed out, (which gets wrapped in `RuntimeException`). If it failed for some other reason then we don't know if the pod is really ready or not. To be clear: I think it's probably OK to carry on with maybe restarting it in either case.